api/test/docker/docker-compose.yaml

```diff
@@ -0,0 +1,183 @@
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+version: "3.6"
+
+services:
+  node1:
+    image: quay.io/coreos/etcd:v3.4.0
+    ports:
+      - "2379:2379"
+    expose:
+      - 2379
+      - 2380
+    networks:
+      apisix_dashboard_e2e:
+        ipv4_address: 172.16.238.100
+    environment:
+      - ETCDCTL_API=3
+    command:
+      - /usr/local/bin/etcd
+      - --data-dir=/etcd-data
+      - --name
+      - node1
+      - --initial-advertise-peer-urls
+      - http://172.16.238.100:2380
+      - --listen-peer-urls
+      - http://0.0.0.0:2380
+      - --advertise-client-urls
+      - http://172.16.238.100:2379
+      - --listen-client-urls
+      - http://0.0.0.0:2379
+      - --initial-cluster
+      - node1=http://172.16.238.100:2380,node2=http://172.16.238.101:2380,node3=http://172.16.238.102:2380
+      - --initial-cluster-state
+      - new
+      - --initial-cluster-token
+      - docker-etcd
+
+  node2:
+    image: quay.io/coreos/etcd:v3.4.0
+    networks:
+      apisix_dashboard_e2e:
+        ipv4_address: 172.16.238.101
+    environment:
+      - ETCDCTL_API=3
+    ports:
+      - "3379:2379"
+    expose:
+      - 2379
+      - 2380
+    command:
+      - /usr/local/bin/etcd
+      - --data-dir=/etcd-data
+      - --name
+      - node2
+      - --initial-advertise-peer-urls
+      - http://172.16.238.101:2380
+      - --listen-peer-urls
+      - http://0.0.0.0:2380
+      - --advertise-client-urls
+      - http://172.16.238.101:2379
+      - --listen-client-urls
+      - http://0.0.0.0:2379
+      - --initial-cluster
+      - node1=http://172.16.238.100:2380,node2=http://172.16.238.101:2380,node3=http://172.16.238.102:2380
+      - --initial-cluster-state
+      - new
+      - --initial-cluster-token
+      - docker-etcd
+
+  node3:
+    image: quay.io/coreos/etcd:v3.4.0
+    networks:
+      apisix_dashboard_e2e:
+        ipv4_address: 172.16.238.102
+    environment:
+      - ETCDCTL_API=3
+    ports:
+      - "4379:2379"
+    expose:
+      - 2379
+      - 2380
+    command:
+      - /usr/local/bin/etcd
+      - --data-dir=/etcd-data
+      - --name
+      - node3
+      - --initial-advertise-peer-urls
+      - http://172.16.238.102:2380
+      - --listen-peer-urls
+      - http://0.0.0.0:2380
+      - --advertise-client-urls
+      - http://172.16.238.102:2379
+      - --listen-client-urls
+      - http://0.0.0.0:2379
+      - --initial-cluster
+      - node1=http://172.16.238.100:2380,node2=http://172.16.238.101:2380,node3=http://172.16.238.102:2380
+      - --initial-cluster-state
+      - new
+      - --initial-cluster-token
+      - docker-etcd
+
+  upstream:
+    image: johz/upstream:v2.0
+    restart: always
+    ports:
+      - '80:80/tcp'
+      - '1980:1980/tcp'
+      - '1981:1981/tcp'
+      - '1982:1982/tcp'
+      - '1983:1983/tcp'
+      - '1984:1984/tcp'
+    networks:
+      apisix_dashboard_e2e:
+        ipv4_address: 172.16.238.120
```

Comment:
Can we use a simpler IP address here? eg: `10.1.1.2**`

Reply:
Ping @nic-chen

Reply:
will update next move.

Reply:
updated.

Reply:
I try to use `10.1.1.*` as docker subnetwork IP, It's ok in my local, but CI fail. so revert that first, maybe try later.